src/lib.rs

```diff
@@ -1234,87 +1290,69 @@ pub const IID_IMMDeviceEnumerator: IID = GUID {
     Data4: [0xA7, 0x46, 0xDE, 0x8D, 0xB6, 0x36, 0x17, 0xE6],
 };
 
-#[repr(C)]
-pub struct IMMDeviceVtbl {
-    pub QueryInterface: unsafe extern "system" fn(
-        This: *mut IMMDevice,
+RIDL!(
+interface IMMDevice(IMMDeviceVtbl) {
```

Comment:
Could you change the COM definitions to also inherit from their base COM classes, like the way I did with `AsyncIUnknown`?